src/runtime-rs/crates/shimv2/src/shim.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::{
+    os::unix::ffi::OsStrExt,
+    path::{Path, PathBuf},
+};
+
+use anyhow::{anyhow, Context, Result};
+
+use crate::{Args, Error};
+
+const SOCKET_ROOT: &str = "/run/containerd";
+const SHIM_SOCKET: &str = "shim.sock";
+
+pub(crate) const ENV_KATA_BIND_FD: &str = "KATA_BIND_FD";
+
+/// Command executor for shim.
+pub struct ShimExecutor {
+    pub(crate) args: Args,
+}
+
+impl ShimExecutor {
+    /// Create a new instance of [`Shim`].
+    pub fn new(args: Args) -> Self {
+        ShimExecutor { args }
+    }
+
+    pub(crate) fn load_oci_spec(&self) -> Result<oci::Spec> {
+        let bundle_path = self.get_bundle_path()?;
+        let spec_file = bundle_path.join("config.json");
+
+        oci::Spec::load(spec_file.to_str().unwrap_or_default()).context("load spec")
+    }
+
+    pub(crate) fn write_address(&self, address: &Path) -> Result<()> {
+        let dir = self.get_bundle_path()?;
+        let file_path = &dir.join("address");
+        std::fs::write(file_path, address.as_os_str().as_bytes())
+            .context(Error::FileWrite(format!("{:?}", &file_path)))
+    }
+
+    pub(crate) fn write_pid_file(&self, pid: u32) -> Result<()> {
+        let dir = self.get_bundle_path()?;
+        let file_path = &dir.join("shim.pid");
+        std::fs::write(file_path, format!("{}", pid))
+            .context(Error::FileWrite(format!("{:?}", &file_path)))
+    }
+
+    pub(crate) fn read_pid_file(&self, bundle_path: &Path) -> Result<u32> {
+        let file_path = bundle_path.join("shim.pid");
+        let data = std::fs::read_to_string(&file_path)
+            .context(Error::FileOpen(format!("{:?}", file_path)))?;
+
+        data.parse::<u32>().context(Error::ParseString)
+    }
+
+    pub(crate) fn get_bundle_path(&self) -> Result<PathBuf> {
+        std::env::current_dir().context(Error::GetBundlePath)
```

Comment:
This looks odd to me. Is it _guaranteed_ that `current_dir` is actually the bundle directory?